packages/core/src/http-middlewares/before/add-digest-auth-header.js

```diff
@@ -15,7 +15,7 @@ const buildDigestHeader = (username, password, url, method, creds) => {
     );
   }
 
-  const path = urllib.URL(url).path;
+  const path = new urllib.URL(url).path;
```

Comment:
Looks like `path` is not a valid attribute of the `URL` object.

```
> const { URL } = require('url')
undefined
> new URL('https://example.com/test/foo').path
undefined
> new URL('https://example.com/test/foo').pathname
'/test/foo'
```

We should use `pathname`:

```suggestion
  const path = new urllib.URL(url).pathname;
```